lib/alephant/lookup/lookup_helper.rb

```diff
@@ -55,6 +59,14 @@ def write(id, opts, batch_version, location)
       def truncate!
         @lookup_table.truncate!
       end
+
+      private
+
+      def component_cache_key(id, opts, batch_version)
+        "#{@lookup_table.table_name}/COMPONENT_KEY/#{batch_version}".gsub("COMPONENT_KEY") do |s|
```

Comment:
@samfrench can change this to something like:

```rb
def component_cache_key(id, opts, batch_version)
  template_key(batch_version).gsub("{{component_key}}") do |s|
    LookupLocation.new(id, opts, batch_version).component_key
  end
end

def template_key(batch_version)
  "#{@lookup_table.table_name}/{{component_key}}/#{batch_version}"
end
```

Just felt there was too much going on in that one method. Also I prefer seeing a more typical template syntax (e.g. Mustache) used when doing interpolation; for consistency reasons more than anything else as that's a well known syntax for template interpolation. I'm sure this would be a contentious argument but that's my opinion on it :-)